src/components/todos/todo-dialog.tsx

```diff
@@ -0,0 +1,77 @@
+import { CreateEditTodoForm, TTodoFormData } from '@/components/todos/create-edit-todo-form'
+import {
+  AlertDialog,
+  AlertDialogContent,
+  AlertDialogHeader,
+  AlertDialogTitle,
+  AlertDialogTrigger,
+} from '@/components/ui/alert-dialog'
+import { type ReactNode } from 'react'
+
+type BaseProps = {
+  open: boolean
+  children: ReactNode
+  onOpenChange: (open: boolean) => void
+}
+
+type CreateModeProps = BaseProps & {
+  mode: 'create'
+  defaultData?: Partial<TTodoFormData>
+  onSubmit: (data: TTodoFormData) => void
+  isMutationPending?: boolean
+}
+
+type EditModeProps = BaseProps & {
+  mode: 'edit'
+  defaultData: Partial<TTodoFormData>
+  onSubmit: (data: TTodoFormData) => void
+  isMutationPending?: boolean
+}
+
+type DialogMode = 'create' | 'edit' | 'view'
+
+type ViewModeProps = BaseProps & {
+  mode: 'view'
+  defaultData: Partial<TTodoFormData>
+}
+
+type TodoDialogProps = CreateModeProps | EditModeProps | ViewModeProps
+
+export const TodoDialog = (props: TodoDialogProps) => {
+  const { mode, open, children, defaultData, onOpenChange } = props
+  const TITLE_BY_MODE: Record<DialogMode, string> = {
+    create: 'Create Todo',
+    edit: 'Edit Todo',
+    view: 'View Todo',
+  }
```

Comment:
- if this is a constant then we can move it outside the `TodoDialog` component to avoid creating the object when the component is re-rendered.